polling_stations/management/commands/drop_failed_replication_slots.py

```diff
@@ -0,0 +1,151 @@
+from datetime import date
+from typing import Dict, Any
+
+from django.core.management.base import BaseCommand
+from django.conf import settings
+
+from core.slack_client import SlackClient
+from settings.constants.slack import BOTS_CHANNEL, BOTS_TESTING_CHANNEL
+from polling_stations.db_routers import (
+    get_principal_db_name,
+    get_principal_db_connection,
+)
+
+DB_NAME = get_principal_db_name()
+
+
+class Command(BaseCommand):
+    help = "Checks for failed replication slots in RDS and drops them, then reports to Slack"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.post_to_slack = False
+        self.dc_environment = None
+        self.cursor = get_principal_db_connection().cursor()
+        self.slack_client = SlackClient(channel=self.slack_channel)
+        self.failed_slots = []
+        self.drop_succeeded = []
+        self.drop_failed = []
+
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument(
+            "--send-slack-report",
+            action="store_true",
+            help="Ask import_eoni command to report to slack",
+        )
+
+    def handle(self, *args, **options):
+        try:
+            self.dc_environment = settings.DC_ENVIRONMENT
+            self.post_to_slack = options.get("post_to_slack")
+            self.check_replication_slots()
+
+            if self.failed_slots:
+                self.drop_failed_slots()
+                self.report_to_slack()
+                return
+
+            if date.today().weekday() == 0:
+                # Report once a week even if no failed slots found, so we know bot is working.
+                self.report_to_slack()
+                return
+
+        except (KeyboardInterrupt, SystemExit):
```

Comment:
This is redundant as both inherit from `BaseException` not `Exception`. See https://docs.python.org/3/library/exceptions.html#exception-hierarchy

Reply:
Fixed in 0067957b2. I cribbed `from polling_stations/apps/data_importers/management/commands/import_eoni_from_s3.py` - but some archaeology suggests that got [added in unnecessarily](https://github.com/DemocracyClub/UK-Polling-Stations/pull/8192#discussion_r1888669924).